### ISSUE TITLE
sessionExpired errors when using the driver from nodeJS to aura

### ISSUE THREAD

**Issue by jroith** [neo4j/neo4j-javascript-driver#1134]
I'm still getting random sessionExpires errors when connecting to aura even though I have set the max connection timeout to 9 minutes. Wouldn't it be better if the driver would transparently retry instead of failing? And in any case, why can it not just retry with a different or even a new connection?

Also it would be nice to maintain the connections in the pool more actively to prevent this problem.

**Comment by bigmontz** [comment]
Hi @jroith,


we have methods in the driver which internally retries when some cluster failures happens. The methods are:

* [Driver.executeQuery](https://neo4j.com/docs/api/javascript-driver/current/class/lib6/driver.js~Driver.html#instance-method-executeQuery)([Manual](https://neo4j.com/docs/javascript-manual/current/#_query_the_database)). This one should be used to run a single query when you don't need to have control over the cursor/stream. 
* [Session.executeRead](https://neo4j.com/docs/api/javascript-driver/current/class/lib6/session.js~Session.html#instance-method-executeRead)([Manual](https://neo4j.com/docs/javascript-manual/current/#_run_your_own_transactions)). For read queries.
* [Session.executeWrite](https://neo4j.com/docs/api/javascript-driver/current/class/lib6/session.js~Session.html#instance-method-executeWrite)([Manual](https://neo4j.com/docs/javascript-manual/current/#_run_your_own_transactions)). For write queries.  

Are you using these methods? Could you share some code samples?

**Comment by jroith** [comment]
This happens from the neo4j GraphQL library and they are using these methods. I will add more logging and try to report more information. Unfortunately the issue is difficult to reproduce and only occurs sometimes.

**Comment by bigmontz** [comment]
@jroith anything new about this?

**Comment by jroith** [comment]
Yes, it's getting worse and feels unpredictable for long-running transactions. I'm currently considering running the community or enterprise version instead of aura.

**Comment by bigmontz** [comment]
Can you share some logs?

**Comment by bigmontz** [comment]
@jroith Does the problem persists? Can you share some log?

**Comment by jroith** [comment]
Not recently. I have just upgraded to 5.17 and will keep monitoring the issue.
